$OpenBSD: patch-src_server_c,v 1.11 2013/11/08 21:39:43 brad Exp $

Fix SNI handling; only use key+cert from SNI specific config. CVE-2013-4508

--- src/server.c.orig	Fri Aug 30 14:46:26 2013
+++ src/server.c	Fri Nov  8 16:17:55 2013
@@ -314,6 +314,9 @@ static void server_free(server *srv) {
 			buffer_free(s->ssl_verifyclient_username);
 #ifdef USE_OPENSSL
 			SSL_CTX_free(s->ssl_ctx);
+			EVP_PKEY_free(s->ssl_pemfile_pkey);
+			X509_free(s->ssl_pemfile_x509);
+			if (NULL != s->ssl_ca_file_cert_names) sk_X509_NAME_pop_free(s->ssl_ca_file_cert_names, X509_NAME_free);
 #endif
 			free(s);
 		}
